nupic/research/frameworks/pytorch/restore_utils.py

```diff
@@ -0,0 +1,147 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2018, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+import io
+import os
+import pickle
+
+import torch
+
+
+def get_state_dict(checkpoint_path):
```

Comment:
We have started gzipping the state dicts. Maybe you should use `deserialize_state_dict` in model_utils.py

Reply:
You should use `deserialize_state_dict` and pass the `device`.
See `set_state` method on `imagenet_experiment.py`